tools/lib/src/publish_command.dart

```diff
@@ -256,51 +256,52 @@ Safe to ignore if the package is deleted in this commit.
   }
 
   void _ensureValidPubCredential() {
-    final String credentialsPath = _credentialsPath;
+    final String credentialsPath =
+        _getCredentialsPath(platform: platform, path: path);
     final File credentialFile = packagesDir.fileSystem.file(credentialsPath);
     if (credentialFile.existsSync() &&
         credentialFile.readAsStringSync().isNotEmpty) {
       return;
     }
-    final String? credential = io.Platform.environment[_pubCredentialName];
+    final String? credential = platform.environment[_pubCredentialName];
     if (credential == null) {
       printError('''
 No pub credential available. Please check if `$credentialsPath` is valid.
 If running this command on CI, you can set the pub credential content in the $_pubCredentialName environment variable.
 ''');
       throw ToolExit(1);
     }
+    credentialFile.createSync(recursive: true);
     credentialFile.openSync(mode: FileMode.writeOnlyAppend)
       ..writeStringSync(credential)
       ..closeSync();
   }
 }
 
-final String _credentialsPath = () {
-  String? cacheDir;
-  final String? pubCache = io.Platform.environment['PUB_CACHE'];
-  if (pubCache != null) {
-    cacheDir = pubCache;
-  } else if (io.Platform.isWindows) {
-    final String? appData = io.Platform.environment['APPDATA'];
-    if (appData == null) {
-      printError('"APPDATA" environment variable is not set.');
-    } else {
-      cacheDir = p.join(appData, 'Pub', 'Cache');
-    }
-  } else {
-    final String? home = io.Platform.environment['HOME'];
+/// The path in which pub expects to find its credentials file.
+String _getCredentialsPath({
+  required Platform platform,
+  required p.Context path,
+}) {
+  // See https://github.com/dart-lang/pub/blob/master/doc/cache_layout.md#layout
+  String? configDir;
+  String? configHome = platform.environment['XDG_CONFIG_HOME'];
+  if (configHome == null) {
+    final String? home = platform.environment['HOME'];
     if (home == null) {
       printError('"HOME" environment variable is not set.');
     } else {
-      cacheDir = p.join(home, '.pub-cache');
+      configHome = path.join(home, '.config');
     }
   }
+  if (configHome != null) {
+    configDir = path.join(configHome, 'dart');
+  }
 
-  if (cacheDir == null) {
-    printError('Unable to determine pub cache location');
+  if (configDir == null) {
+    printError('Unable to determine pub con location');
```

Comment:
I think `config` or `conf` is helpful for understanding.

```suggestion
    printError('Unable to determine pub config location');
```

Reply:
I noticed but I wanted to keep the original code as it is :)

https://github.com/flutter/packages/blob/4a5cc3e6947b2b9840d7d810fe310e00726803fb/script/tool/lib/src/publish_command.dart#L459

Reply:
OK :)